code/controllers/subsystem/spatial_gridmap.dm

```diff
@@ -0,0 +1,256 @@
+/datum/spatial_grid_cell
+	///our x index in the list of cells
+	var/cell_x
+	///our y index in the list of cells
+	var/cell_y
+	///what z level we are in
+	var/cell_z
+
+	//every data point in a grid cell is separated by usecase
+
+	///every hearing sensitive movable inside this cell
+	var/list/hearing_contents = list()
+	///every client possessed mob inside this cell
+	var/list/client_contents = list()
```

Comment:
define generically

Reply:
Rather then a new list per type of contents, a list of lists indexed by type.
I guess what's important is how easily it to be for people to add new categories. Reading through the comments it sounds like the answer to that is reasonably hard

Reply:
Also this should maybe be a lazylist

Reply:
i dont think it should be made into a single list, making associative indexes is very expensive and movement is already a massive drain on performance, most of that being player movement. a player moving from a cell containing only themselves to an empty cell is making and deleting one big list, and two associative indexes and two sub lists as opposed to adding to two lists currently. not to mention it will slow reads.

if someone wants to add a new content type they should be able to do so without compromising performance in the slightest, especially if more systems start to rely on the grid. its essentially just adding their new content define to the gridmap defines folder then adding it to the switch statements (as well as the list correlated with that define). i should probably make the switch statements a define though

Reply:
also the content lists are now lazy

Reply:
pogged up